Firestore/Swift/Source/Result/CollectionReference+Result.swift

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+@available(swift 5.0)
+extension CollectionReference {
+  /// Adds a new document to this collection with the specified data, assigning it a document ID
+  /// automatically.
+  ///
+  /// - Parameters:
+  ///   - data: The `Dictionary` containing the data for the new document.
+  ///   - completion: The closure to execute on successfully writing to the server or on receipt of
+  ///   an error. It will not be called while the client is offline, though local changes will be visible
+  ///   immediately.
+  ///   - result: The result of request. On success it is empty, otherwise it contains an `Error`.
+  /// - Returns: A `DocumentReference` pointing to the newly created document.
+  @discardableResult
+  func addDocument(data: [String: Any],
+                   completion: @escaping (_ result: Result<Void, Error>) -> Void)
+    -> DocumentReference {
+    addDocument(data: data, completion: mapResultCompletion(completion))
```

Comment:
This fails to compile in Xcode 10.3, which is our best proxy for a Swift 5.0 compiler that's not actually Swift 5.1. See https://travis-ci.org/github/firebase/firebase-ios-sdk/jobs/744788354.

This needs an explicit return statement. Also in the overload below.

Reply:
```suggestion
    return addDocument(data: data, completion: mapResultCompletion(completion))
```